### DISCUSSION TITLE
Add support for query parameter serialization styles

### DISCUSSION THREAD

**Discussion by jakub-bacic** [fastapi/fastapi#8940] [Questions]
While implementing filtering API recently and I came across a problem with parsing list of comma-separated ids from query string:

**Example:**
```
GET /api/items?id=1,3,5
```

I know this type of parsing can be easily done by declaring `id` parameter as `str` and parsing it inside the function but then we're missing awesome OpenAPI doc features.

Please find the PR https://github.com/tiangolo/fastapi/pull/2078 with proposed solution.

**OpenAPI Docs:** https://swagger.io/docs/specification/serialization/

**Comments regarding code changes:**
- all changes should be backward compatible
- there's no support for `style=deepObject` as it would require more changes (because query params cannot be non-scalar values at the moment)
- in case multiple values are present (e.g. `?id=1&id=2,3`) and `explode=False`, only last value is parsed (in this case `2,3`); it's consistent with the currently implemented behaviour in case of declaring parameter as non-sequence (e.g. `str`) and receiving more than one value in query string

**Comment by cmflynn** [comment]
Any movement on this?

**Comment by ycd** [comment]
@codyflynnNC OpenAPI specification allows this as Jakub mentioned, and he already opened a PR for this, so for now we are just waiting for @tiangolo's feedback.

**Comment by adriangb** [comment]
This would be a great addition. There is obviously no singular way to do things, and a related problem is getting something like `GET /api/items?id=gt:1,lt:3`. But it would be nice to have _a_ way to do the simplest version of splitting.

**Comment by shufflebits** [comment]
It seems that the reticence on allowing for encoding styles for `deepObject` might be that there is no _standard_ way of encoding it? 
How about allowing for pluggable parameter encoding styles, so we can implement whatever is needed. As it stands, those of us trying to implement endpoints for populating tables that support filtering find it impossible to create a nice documented API that allows for `lt`, `gt` and so on comparison operators.
